WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/quickstart/QuickStartCardSource.kt

```diff
@@ -21,7 +21,7 @@ class QuickStartCardSource @Inject constructor(
     private val quickStartUtilsWrapper: QuickStartUtilsWrapper,
     private val selectedSiteRepository: SelectedSiteRepository
 ) : MySiteRefreshSource<QuickStartUpdate> {
-    override val refresh = MutableLiveData(true)
+    override val refresh = MutableLiveData(false)
```

Comment:
1. Question (❓): As we are only calling `refresh()` during `build(...)` if a specific condition is met, will that work every time? 
2. Question (❓): Unrelated to this class, but how about `CurrentAvatarSource` and setting the `refresh` live data to `false` there as well, then using the `refresh()` instead of `result.refreshData()`. Is that not updated as well for a reason?

Reply:
1. Actually, no changes are made to the `build(...)` invocation, it is still invoked [every time a site is selected](https://github.com/wordpress-mobile/WordPress-Android/blob/8af489104c5849661423f0070b54e9a6861df192/WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt#L154-L158).  The build function previously called `refreshData(...)` internally, we removed this call and updated the `refresh` `MediatorLiveData` value using `refresh() `call which triggers `onChanged` callback `result.addSource(refresh) { result.refreshData(...) }.` thus calling the `refreshData(...)` function as before.

    The only change here is that every time `build(...)` is invoked, `refresh` value will have the correct initial value `true` to show a loading indicator while the observing callback action is in progress. 

    Also, note that previously, on the initial fragment resume, as the screen was just built, we restricted full refresh `on FirstResume` (which internally called `MySiteRefreshSource.refresh()`).  We've expanded this condition to include site switch events as well because the screen has been just built for the selected site. 

   TL;DR:  
   - `build(...)` is still invoked every time a site is selected which internally invokes`refeshData(...)`.
   - full refresh is restricted on fragment resume whenever `build(...)` is already executed and not just on an initial resume. 

   Let me know if it makes sense or I misunderstood your question. 😄 

2. `CurrentAvatarSource` is a special "site-independent" source, while the issue occurred only when a site is switched so I didn't update it. For consistency, I've updated it in 3d5ced2.

Reply:
👋 @ashiagr !

1. Thanks so much for this elaborate explanation! 🙏 
2. Thank you for updating, even if that was for consistency purposes only! 🙏